### DISCUSSION TITLE
100% discounted orders aren't marked as paid

### DISCUSSION THREAD

**Discussion by virajsoni06** [swellstores/community#474] [Help] [answered]
Hi, in our store we have internal coupons which gives 100% discount. However when those are applied, the orders aren't marked as paid and we cannot fetch completed orders for that customers. This seems like a bug in the system

**Comment by steven-k1** [comment]
@virajsoni06 Looking into this

**Comment by steven-k1** [comment]
@virajsoni06 Looked into this and found that this is a known issue, but a fix would not be able to be implemented at this time as it would affect many users and could have negative effects depending on various use cases. We plan to add model versioning, which would allow us to apply a fix for this then, but there is no ETA at this time.

In the meantime, this can be fixed directly within your account if you wish to do so.

You will want to navigate to your developer console within the admin dashboard and run the following request.

- GET /:models /com.orders/fields/paid

- Copy the formula field and update `payment_total > 0` to `payment_total >= 0` within the formula, and then use it in the query below

- PUT /:models /com.orders/fields/paid
```
{
 "formula": "boolean(and(not(draft), not(canceled), or(payment_marked, and(payment_total >= 0, payment_balance >= 0))))"
}
```

**Reply by ericingram** [reply, marked as answer]
To clarify, it's not considered a bug as it was designed to work this way intentionally. Only when payments are made is an order considered "paid". The issue is that subscriptions were designed differently and we would like to make the two models consistent, but cannot make the design change while many users are expecting it as such.

**Comment by virajsoni06** [comment]
Hi @steven-k1 we updated the model and it still is the same, the order is not marked as paid

**Reply by steven-k1** [reply]
@virajsoni06 Are you referring to the existing order?

If not already done, are you able to place a test order to confirm if this is reflected on new orders?

**Reply by virajsoni06** [reply]
@steven-k1 referring to new orders. Placed a new order and still wasn't marked as paid

**Reply by steven-k1** [reply]
@virajsoni06 Looking into it

**Reply by steven-k1** [reply]
@virajsoni06 I spoke with our team to clarify the workaround provided and what the workaround does is mark the orders as paid within the API so the order is processed as normal, but this does not change the visual status that you see within the dashboard.

The Swell admin has its own conditions for displaying the order status and would require a code change for the visual to change, but as pointed out in the reply from @ericingram above, we cannot make the design change while many users are expecting it as such.

What I can do though, is log a request for an improvement to be made for this to be discussed with our team.
